post-init/master-post-init.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+#
+# Can be used either under --metadata startup-script-url=... or with
+# --initialization-actions to invoke commands (like submitting a job)
+# after cluster becomes initially healthy and then on every master
+# reboot or a single time on cluster deployment, respectively.
+#
+# Examples:
+#   Submit a single job with cluster and turn off cluster on completion:
+#
+#     gcloud dataproc clusters create ${CLUSTER_NAME} --scopes cloud-platform --initialization-actions gs://${BUCKET}/master-post-init.sh --metadata post-init-command="gcloud dataproc jobs submit hadoop --cluster ${CLUSTER_NAME} --jar file:///usr/lib/hadoop-mapreduce/hadoop-mapreduce-examples.jar teragen 10000 /tmp/teragen; gcloud dataproc clusters delete -q ${CLUSTER_NAME}"
+#
+#   Create a cluster which runs a "forever-running" job, resubmitting on reboot
+#   or if the job terminates for any reason:
+#
+#     gcloud dataproc clusters create ${CLUSTER_NAME} --scopes cloud-platform --metadata startup-script-url=gs://${BUCKET}/master-post-init.sh,post-init-command="while true; do gcloud dataproc jobs submit spark --cluster ${CLUSTER_NAME} --jar gs://${BUCKET}/my-longlived-job.jar foo args; done"
+
+METADATA_ROOT='http://metadata/computeMetadata/v1/instance/attributes'
+
+# Only run on master. If you want to run on all nodes instead, then simply
+# remove these lines which exit for non-master nodes.
+ROLE=$(curl -f -s -H Metadata-Flavor:Google ${METADATA_ROOT}/dataproc-role)
+if [[ "${ROLE}" != 'Master' ]]; then
+  exit 0
+fi
+
+CLUSTER_NAME=$(curl -f -s -H Metadata-Flavor:Google \
+    ${METADATA_ROOT}/dataproc-cluster-name)
```

Comment:
The init actions have partially switched to this, but I guess we can go either way if we're going to add compatibility wrappers anyhow.